pkg/alexa/gen/skill.go

```diff
@@ -6,415 +6,283 @@ import (
-	return Intent{Name: name}
+// TODO: add WithLocale(Description|Summary|...)
+func (l *SkillLocaleBuilder) WithSummary(summary string) *SkillLocaleBuilder {
+	l.skillSummary = summary
+	return l
 }
-
-func (i *Intent) AddSlot(slot Slot) {
-	i.Slots = append(i.Slots, slot)
+func (l *SkillLocaleBuilder) WithExamples(examples string) *SkillLocaleBuilder {
+	l.skillExamples = examples
+	return l
 }
-
-// Slot
-type Slot struct {
-	Name         string // specific to intent
-	Type         Type   // global for skill
-	Confirmation bool
-	Elicitation  bool
+func (l *SkillLocaleBuilder) WithKeywords(keywords string) *SkillLocaleBuilder {
+	l.skillKeywords = keywords
+	return l
 }
-
-func NewSlot(name string, t Type) Slot {
-	return Slot{Name: name, Type: t, Confirmation: false, Elicitation: false}
+func (l *SkillLocaleBuilder) WithSmallIcon(smallicon string) *SkillLocaleBuilder {
+	l.skillSmallIcon = smallicon
+	return l
 }
-
-// TODO: overkill? just use 'string'?
-// Type
-type Type string
-
-func NewType(t string) Type {
-	return Type(t)
+func (l *SkillLocaleBuilder) WithLargeIcon(largeicon string) *SkillLocaleBuilder {
+	l.skillLargeIcon = largeicon
+	return l
 }
-
-// TODO: what for?
-// Model
-type Model struct {
-	Model *alexa.Model
+func (l *SkillLocaleBuilder) WithPrivacyURL(privacy string) *SkillLocaleBuilder {
+	l.skillPrivacyURL = privacy
+	return l
 }
-
-// Flags for alexa.Privacy
-const (
-	FlagIsExportCompliant string = "IsExportCompliant"
-	FlagContainsAds       string = "ContainsAds"
-	FlagAllowsPurchases   string = "AllowsPurchases"
-	FlagUsesPersonalInfo  string = "UsesPersonalInfo"
-	FlagIsChildDirectred  string = "IsChildDirected"
-)
-
-// privacy
-type privacy struct {
-	flags map[string]bool
+func (l *SkillLocaleBuilder) WithTermsURL(terms string) *SkillLocaleBuilder {
+	l.skillTermsURL = terms
+	return l
 }
 
-func (s *SkillBuilder) SetPrivacyFlag(name string, b bool) {
-	s.privacy.flags[name] = b
+func (l *SkillLocaleBuilder) BuildPublishingLocale() (alexa.LocaleDef, error) {
+	if loc, err := l.registry.Resolve(l.locale); err != nil {
+		return alexa.LocaleDef{}, err
+	} else {
+		return alexa.LocaleDef{
+			Name:         loc.Get(l.skillName),
+			Description:  loc.Get(l.skillDescription),
+			Summary:      loc.Get(l.skillSummary),
+			Keywords:     loc.GetAll(l.skillKeywords),
+			Examples:     loc.GetAll(l.skillExamples),
+			SmallIconURI: loc.Get(l.skillSmallIcon),
+			LargeIconURI: loc.Get(l.skillLargeIcon),
+		}, nil
+	}
 }
 
-// TODO implement other functions
-
-// prompt is used internally to build the models
-type prompt struct {
-	Id            string
-	Variations    []alexa.PromptVariations
-	Confirmations []alexa.PromptVariations
+func (l *SkillLocaleBuilder) BuildPrivacyLocale() (alexa.PrivacyLocaleDef, error) {
+	if loc, err := l.registry.Resolve(l.locale); err != nil {
+		return alexa.PrivacyLocaleDef{}, err
+	} else {
+		return alexa.PrivacyLocaleDef{
+			PrivacyPolicyURL: loc.Get(l.skillPrivacyURL),
```

Comment:
According to skill schema its per locale, would make sense to me (but then again not really) because Privacy etc. is of course localized, but I would rather make the webserver return the locale based on preferred browser language anyway.